CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.cpp

```diff
@@ -0,0 +1,571 @@
+    }
+
+    // Ask the user if they want to commit or rollback the transaction
+    emit commitDialogRequested();
+}
+
+void EditGeodatabaseWithTransactions::commitTransaction()
+{
+    if (!m_geodatabase)
+    {
+        return;
+    }
+
+    // Commit the transaction to store the edits (this will also end the transaction)
+    m_geodatabase->commitTransaction();
+    setMessageText("Edits committed to geodatabase.");
+}
+
+void EditGeodatabaseWithTransactions::rollbackTransaction()
+{
+    if (!m_geodatabase)
+    {
+        return;
+    }
+
+    // Rollback the transaction to discard the edits (this will also end the transaction)
+    m_geodatabase->rollbackTransaction();
+    setMessageText("Edits discarded.");
+}
+
+void EditGeodatabaseWithTransactions::cancelTransaction()
+{
+    // User canceled - keep transaction active
+    setMessageText("Transaction started.");
+}
+
+void EditGeodatabaseWithTransactions::updateFeatureTypesForTable(GeodatabaseFeatureTable* table)
+{
+    if (!table)
+    {
+        return;
+    }
+
+    m_currentFeatureTypes.clear();
+
+    QStringList featureTypeNames;
+    for (const FeatureType& featureType : table->featureTypes())
+    {
+        featureTypeNames.append(featureType.name());
+    }
+
+    featureTypeNames.sort(Qt::CaseInsensitive);
+    m_currentFeatureTypes = featureTypeNames;
+
+    emit currentFeatureTypesChanged();
+}
+
+void EditGeodatabaseWithTransactions::setSelectedTableName(const QString& tableName)
+{
+    if (m_selectedTableName != tableName)
+    {
+        m_selectedTableName = tableName;
+        emit selectedTableNameChanged();
+
+        // Update feature types for the newly selected table
+        GeodatabaseFeatureTable* table = m_tablesByName.value(tableName, nullptr);
+        if (table)
+        {
+            updateFeatureTypesForTable(table);
+        }
+    }
+}
+
+void EditGeodatabaseWithTransactions::addFeatureAtLocation(const QString& tableName, const QString& featureTypeName, const QPoint& location)
+{
+    if (!m_mapView || !m_geodatabase)
+    {
+        return;
+    }
+
+    // Get the correct table using dynamic lookup
+    GeodatabaseFeatureTable* targetTable = m_tablesByName.value(tableName, nullptr);
+
+    if (!targetTable)
+    {
+        setMessageText(QString("Table '%1' not found.").arg(tableName));
+        return;
+    }
+
+    // Find the feature type
+    FeatureType selectedFeatureType;
+    bool foundType = false;
+    for (const FeatureType& featureType : targetTable->featureTypes())
+    {
+        if (featureType.name() == featureTypeName)
+        {
+            selectedFeatureType = featureType;
+            foundType = true;
+            break;
+        }
+    }
+
+    if (!foundType)
+    {
+        setMessageText(QString("Feature type '%1' not found in table '%2'.").arg(featureTypeName, tableName));
+        return;
+    }
+
+    // Convert screen point to map point
+    Point mapPoint = m_mapView->screenToLocation(location.x(), location.y());
```

Comment:
Although geometries are immutable, it is still best practices to make everything possible const.

```suggestion
    const Point mapPoint = m_mapView->screenToLocation(location.x(), location.y());
```